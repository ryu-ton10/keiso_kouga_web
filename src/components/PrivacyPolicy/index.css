.overlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.5);

  display: flex;
  align-items: center;
  justify-content: center;
}

.overlay .modal {
  background: #f5f5f5;
  padding: 20px 10px 20px 10px;
  border-radius: 3px;
  max-width: 70%;
  text-align: center;
}

.overlay .modal button {
  margin: 10px 0 10px 0;
  font-size: 15pt;
  color: white;
  background-color: #00bfff;
  border-color: white;
}

.overlay .modal p {
  font-size: 12pt;
  color: #808080;
}
