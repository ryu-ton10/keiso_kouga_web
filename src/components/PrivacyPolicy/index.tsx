import './index.css';

type Props = {
  scentence: string;
  hook: (args: boolean) => void;
}

function PrivacyPolicy(props: Props) {

  const { scentence, hook } = props;

  return(
    <div className="overlay">
      <div className="modal">
        <p>{scentence}</p>
        <button type="button" onClick={ () => hook(false) }>OK</button>
      </div>
    </div>
  );
};

export default PrivacyPolicy;
