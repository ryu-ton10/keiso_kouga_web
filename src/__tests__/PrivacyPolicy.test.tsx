import renderer from "react-test-renderer";
import PrivacyPolicy from "components/PrivacyPolicy";

const dummyScentence = "hogehoge";
const dummyHook = jest.fn();

/* ---------------- snapshot test -------------------- */
it("コンポーネントが描画されること", () => {
  const tree = renderer
    .create(<PrivacyPolicy scentence={dummyScentence} hook={dummyHook} />)
    .toJSON();

  expect(tree).toMatchInlineSnapshot(`
    <div
      className="overlay"
    >
      <div
        className="modal"
      >
        <p>
          hogehoge
        </p>
        <button
          onClick={[Function]}
        >
          OK
        </button>
      </div>
    </div>
  `);
});
