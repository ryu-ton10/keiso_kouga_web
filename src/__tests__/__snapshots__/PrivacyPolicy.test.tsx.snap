// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`コンポーネントが描画されること 1`] = `
<div
  className="overlay"
>
  <div
    className="modal"
  >
    <p>
      hogehoge
    </p>
    <button
      onClick={[Function]}
      type="button"
    >
      OK
    </button>
  </div>
</div>
`;
